swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -207,10 +244,10 @@ private void persistContract(ContractEntity entity, OffsetDateTime now) {
   }
 
   private Optional<ContractEntity> currentlyActiveContract(ContractEntity contract) {
-    Map<String, Object> stringObjectMap =
-        Map.of("subscriptionNumber", contract.getSubscriptionNumber());
-
-    return contractRepository.getContract(stringObjectMap, true);
```

Comment:
This isn't duplicate code. :)
I guess you need multiple contracts but, for upstream I need single contract by subscriptionnumber and end date null. Before we delete this lets make sure that is accounted.

Reply:
I think we have the equivalent here:

https://github.com/RedHatInsights/rhsm-subscriptions/pull/1930/files/f6f6c460c7532f1284bb8f43ca3eec30ba4641bc#diff-d0a3f94bef59f416c81d8ea1f761cdfcd6e5ae3e5888e580e5b398772ce8b5f1R247